redash/query_runner/snowflake.py

```diff
@@ -146,10 +146,12 @@ def _parse_results(self, cursor):
            row_dict = {}
            for column, value in zip(cursor.description, row):
                column_name = self._column_name(column[0])
-               if column[1] == 0:
-                   row_dict[column_name] = str(value)  # Cast INT to STRING to avoid potential overflow in JS
-               else:
+               # Same as _determine_type, we need to check the data type [1] and scale [5].
+               if column[1] == 0 and column[5] > 0:
                    row_dict[column_name] = value
+               else:
+                   # Cast INT to STRING to avoid potential overflow in JS
+                   row_dict[column_name] = str(value)
```

Comment:
Doesn't this change all values types that aren't `number` to strings?
```suggestion
               if column[1] == 0 and column[5] == 0:
                   # Cast INT to STRING to avoid potential overflow in JS
                   row_dict[column_name] = str(value)
               else:
                   row_dict[column_name] = value
```

Reply:
I just pushed the wrong commit...